# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1525119478 18000
#      Mon Apr 30 15:17:58 2018 -0500
# Node ID af64921cdb2898dac6440790a1c73bd2e8ae621c
# Parent  cea0b7e47a0e7de07f294052ee970c5b63aad7ac
Make figure/waitbar window title behavior consistent among toolkits (bug #53793)

* graphics.cc (get_title): Condition on there being a valid number as well
  as 'numbertitle' true to use that number in title.  If title is empty, make
  it a blank " " to avoid any toolkit default window title appearing.

* __gnuplot_drawnow__.m (gnuplot_set_term): Ditto.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -4673,29 +4673,37 @@ figure::properties::update_units (const 
 %!   close (hf);
 %!   set (0, "units", old_units);
 %! end_unwind_protect
 */
 
 std::string
 figure::properties::get_title (void) const
 {
-  if (is_numbertitle ())
+  std::string title;
+  if (! get_number ().isempty () && is_numbertitle ())
     {
       std::ostringstream os;
       std::string nm = get_name ();
 
       os << "Figure " << __myhandle__.value ();
       if (! nm.empty ())
         os << ": " << get_name ();
 
-      return os.str ();
+      title = os.str ();
     }
   else
-    return get_name ();
+    title = get_name ();
+
+  // Qt will use QCoreApplication name (set in main-window.cc)
+  // if the name is empty, so force blank.
+  if (title.empty ())
+    title = " ";
+
+  return title;
 }
 
 octave_value
 figure::get_default (const caseless_str& name) const
 {
   octave_value retval = default_properties.lookup (name);
 
   if (retval.is_undefined ())
diff --git a/scripts/plot/util/__gnuplot_drawnow__.m b/scripts/plot/util/__gnuplot_drawnow__.m
--- a/scripts/plot/util/__gnuplot_drawnow__.m
+++ b/scripts/plot/util/__gnuplot_drawnow__.m
@@ -141,25 +141,25 @@ function enhanced = gnuplot_set_term (pl
     endif
 
     if (! isempty (h) && isfigure (h))
 
       ## Generate gnuplot title string for plot windows.
       if (output_to_screen (term) && ! strcmp (term, "dumb"))
         fig.numbertitle = get (h, "numbertitle");
         fig.name = strrep (get (h, "name"), '"', '\"');
-        if (strcmp (get (h, "numbertitle"), "on"))
+        if (! isempty (get (h, "number")) && strcmp (get (h, "numbertitle"), "on"))
           title_str = sprintf ("Figure %d", h);
+          if (! isempty (fig.name))
+            title_str = sprintf ("%s: %s", title_str, fig.name);
+          endif
+        elseif (! isempty (fig.name))
+          title_str = fig.name;
         else
-          title_str = "";
-        endif
-        if (! isempty (fig.name) && ! isempty (title_str))
-          title_str = sprintf ("%s: %s", title_str, fig.name);
-        elseif (! isempty (fig.name) && isempty (title_str))
-          title_str = fig.name;
+          title_str = " ";
         endif
         if (! isempty (title_str))
           title_str = sprintf ('title "%s"', title_str);
         endif
         if (strcmp (term, "aqua"))
           ## Adjust axes-label and tick-label spacing.
           opts_str = sprintf ('%s font "%s,%d"', opts_str,
                               get (0, "defaultaxesfontname"),
